src/libraries/Microsoft.Extensions.FileSystemGlobbing/src/Abstractions/DirectoryInfoWrapper.cs

```diff
@@ -35,7 +35,17 @@ public override IEnumerable<FileSystemInfoBase> EnumerateFileSystemInfos()
         {
             if (_directoryInfo.Exists)
             {
-                foreach (FileSystemInfo fileSystemInfo in _directoryInfo.EnumerateFileSystemInfos("*", SearchOption.TopDirectoryOnly))
+                IEnumerable<FileSystemInfo> fileSystemInfos;
+                try
+                {
+                    fileSystemInfos = _directoryInfo.EnumerateFileSystemInfos("*", SearchOption.TopDirectoryOnly);
+                }
+                catch (DirectoryNotFoundException)
+                {
+                    yield break;
```

Comment:
> _Confirmed the behavior that we only throw on the root handle_

If I'm getting this correctly, from your comment in the main section, we only expect an exception if specifically the root is not found? It won't happen if a subdirectory is the one that disappears while enumerating?

Reply:
> It won't happen if a subdirectory is the one that disappears while enumerating?

Correct, I linked to the places in DirectoryInfo that handle these cases and differ for subdirectories of a call to enumerate.